src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -51,7 +51,16 @@ public function loadUserByUsername($username)
             $user = $repository->findOneBy(array($this->property => $username));
         } else {
             if (!$repository instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('The Doctrine repository "%s" must implement UserProviderInterface.', get_class($repository)));
+                $message = 'You must either set the "property" option in the corresponding entity provider configuration ';
+                $repositoryClass = get_class($repository);
+
+                if ('Doctrine\ORM\EntityRepository' === $repositoryClass) {
```

Comment:
Right... changed to

> You must either create a Doctrine Repository implementing "Symfony\Component\Security\Core\User\UserProviderInterface" for the entity [entity] or set the "property" option in the corresponding entity provider configuration

only, as we explicitly set the entity class/alias in the provider config (not the repository) it should make easier to guess which provider is concerned I think. WDYT?